### PR TITLE
Add the "toggle visibility" button for articles and content elements again

### PR DESCRIPTION
**Pull request by aschempp**
I noticed the toggle visibility icon for articles was no longer available in the content element view. Which is because `tl_article` now has auto-generated operations, and the toggle operation was missing the `showInHeader` property.

After adding, I got an exception because the new `DataContainerOperation` object was not supported by `DataContainer::generateHeaderOperations` 🙄

### CODE REVIEW

File: core-bundle/contao/classes/DataContainer.php [hunk 1]
```diff
@@ -1173,22 +1146,53 @@ protected function generateHeaderButtons($arrRow, $strPtable)
...
 			if (\is_array($v['button_callback'] ?? null))
 			{
-				$return .= System::importStatic($v['button_callback'][0])->{$v['button_callback'][1]}($arrRow, $v['href'], $label, $title, $v['icon'], $attributes, $strPtable, array(), null, false, null, null, $this);
-				continue;
+				$callback = System::importStatic($v['button_callback'][0]);
+				$ref = new \ReflectionMethod($callback, $v['button_callback'][1]);
+
+				if ($ref->getNumberOfParameters() === 1 && ($type = $ref->getParameters()[0]->getType()) && $type->getName() === DataContainerOperation::class)
+				{
+					$callback->{$v['button_callback'][1]}($config);
```
Comment (leofeyer) [inline core-bundle/contao/classes/DataContainer.php:1159]: Why no `$return .= ` here?

File: core-bundle/contao/classes/DataContainer.php [hunk 2]
```diff
@@ -1197,14 +1201,14 @@ protected function generateHeaderButtons($arrRow, $strPtable)
...
 				}
 
-				$href = System::getContainer()->get('router')->generate($v['route'], $params);
+				$href = System::getContainer()->get('router')->generate($config['route'], $params);
 			}
-			else
+			elseif (isset($config['href']))
 			{
-				$href = $this->addToUrl($v['href'] . '&amp;id=' . $arrRow['id'] . (Input::get('nb') ? '&amp;nc=1' : '') . ($isPopup ? '&amp;popup=1' : ''));
+				$href = $this->addToUrl(($config['href'] ?? '') . '&amp;id=' . $arrRow['id'] . (Input::get('nb') ? '&amp;nc=1' : '') . ($isPopup ? '&amp;popup=1' : ''));
```
Comment (leofeyer) [inline core-bundle/contao/classes/DataContainer.php:1208]: Why `($config['href'] ?? '')`? The value is always set due to the `elseif (isset($config['href']))` above.

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Now `generateButtons()` and `generateHeaderButtons()` both support `DataContainerOperation` objects, but `generateGlobalButtons()` still doesn‘t. Do we need a follow-up PR to add it there as well?

**Comment by leofeyer** [comment]
Thank you @aschempp.

**Comment by aschempp** [comment]
> Now `generateButtons()` and `generateHeaderButtons()` both support `DataContainerOperation` objects, but `generateGlobalButtons()` still doesn‘t. Do we need a follow-up PR to add it there as well?

I don't think the `DataContainerOperation` is compatible with that (at least it currently is not intended to be).

**Pull request merged**